drivers/dma/dma_andes_atcdmac300.c

```diff
@@ -0,0 +1,435 @@
+				current_block->next_block = NULL;
+			}
+		}
+	} else {
+		/* Single transfer is supported, but Chain transfer is still
+		 * not supported. Therefore, set LLPointer to zero
+		 */
+		OUTWORD(DMA_CH_LL_PTR_L(channel), 0);
+		OUTWORD(DMA_CH_LL_PTR_H(channel), 0);
+	}
+
+end:
+	k_spin_unlock(&data->chan[channel].lock, key);
+	return ret;
+}
+
+static int dma_atcdmac300_reload(const struct device *dev, uint32_t channel,
+				 uint32_t src, uint32_t dst, size_t size)
+{
+	struct dma_atcdmac300_data *const data = DEV_DATA(dev);
+	uint32_t src_width;
+	k_spinlock_key_t key;
+
+	if (channel >= ATCDMAC100_MAX_CHAN) {
+		return -EINVAL;
+	}
+
+	key = k_spin_lock(&data->chan[channel].lock);
+
+	/* Set source and destination address */
+	OUTWORD(DMA_CH_SRC_ADDR_L(channel), src);
+	OUTWORD(DMA_CH_SRC_ADDR_H(channel), 0);
+	OUTWORD(DMA_CH_DST_ADDR_L(channel), dst);
+	OUTWORD(DMA_CH_DST_ADDR_H(channel), 0);
+
+	src_width = INWORD(DMA_CH_CTRL(channel)) & DMA_CH_CTRL_SWIDTH_MASK;
+	src_width = (1 << (src_width >> DMA_CH_CTRL_SWIDTH_POS));
+
+	/* Set transfer size */
+	OUTWORD(DMA_CH_TRANSIZE(channel), size/src_width);
+
+	k_spin_unlock(&data->chan[channel].lock, key);
+
+	return 0;
+}
+
+static int dma_atcdmac300_transfer_start(const struct device *dev,
+					 uint32_t channel)
+{
+	struct dma_atcdmac300_data *const data = DEV_DATA(dev);
+	k_spinlock_key_t key;
+
+	if (channel >= ATCDMAC100_MAX_CHAN) {
+		return -EINVAL;
+	}
+
+	key = k_spin_lock(&data->chan[channel].lock);
+
+	SET_MASK(DMA_CH_CTRL(channel), DMA_CH_CTRL_ENABLE);
+
+	data->chan[channel].status.busy = true;
+
+	k_spin_unlock(&data->chan[channel].lock, key);
+
+	return 0;
+}
+
+static int dma_atcdmac300_transfer_stop(const struct device *dev,
+					uint32_t channel)
+{
+	struct dma_atcdmac300_data *const data = DEV_DATA(dev);
+	k_spinlock_key_t key;
+
+	if (channel >= ATCDMAC100_MAX_CHAN) {
+		return -EINVAL;
+	}
+
+	key = k_spin_lock(&data->chan[channel].lock);
+
+	OUTWORD(DMA_ABORT, (1 << channel));
+	OUTWORD(DMA_CH_CTRL(channel), 0);
+	OUTWORD(DMA_INT_STATUS, ((1 << channel) << DMA_INT_STATUS_ABORT_POS));
+
+	data->chan[channel].status.busy = false;
+
+	struct chain_block *current_block = head_block[channel]->next_block;
+
+	while (head_block[channel] != NULL) {
+		k_heap_free(&dma_chain, head_block[channel]);
+		head_block[channel] = current_block;
+		current_block = head_block[channel]->next_block;
+	}
+
+	k_spin_unlock(&data->chan[channel].lock, key);
+
+	return 0;
+}
+
+static int dma_atcdmac300_init(const struct device *dev)
+{
+	const struct dma_atcdmac300_cfg *const config = DEV_CFG(dev);
+	uint32_t ch_num;
+
+	/* Disable all channels and Channel interrupts */
+	for (ch_num = 0; ch_num < ATCDMAC100_MAX_CHAN; ch_num++) {
+		OUTWORD(DMA_CH_CTRL(ch_num), 0);
+	}
+
+	OUTWORD(DMA_INT_STATUS, 0xFFFFFF);
+
+	/* Configure interrupts */
+	config->irq_config();
+
+	irq_enable(config->irq_num);
+
+	return 0;
+}
+static int dma_atcdmac300_get_status(const struct device *dev,
```

Comment:
Nit: missing newline before this.

Reply:
Oops, thank you, we have fixed this problem.